packages/shim-deno/src/deno/stable/functions/open.test.ts

```diff
@@ -0,0 +1,27 @@
+import { open } from './open.ts'
+import assert from "assert/strict";
+import path from 'path';
+
+
+const tempDir = await Deno.makeTempDir();
+
+Deno.test('creates file when createNew is set to true and file does not exist', async () => {
+
+    const filePath = path.join(tempDir, 'some')
+
+    const fileHandle = await open(filePath, { createNew: true, write: true });
+    fileHandle.close();
+
+    assert.ok(await Deno.stat(filePath));
+});
+
+Deno.test('errors when createNew is set to true and file exists', async () => {
+
+    const testFile = await Deno.makeTempFile({ dir: tempDir });
+
+    await assert.rejects(async () => {
+        await open(testFile, { createNew: true, write: true });
+    }, { message: `EEXIST: file already exists, open '${testFile}'` });
+});
+
+Deno.remove(tempDir, { recursive: true });
```

Comment:
This will occur before the tests run so we can't do this here. I think maybe we should move the `await Deno.makeTempDir()` calls into each test and have them clean it up individually. This could be achieved by adding a utility function like so:

```ts
async function withTempDir(test: (tempDirPath: string) => Proimse<void>) {
  const tempDirPath = await Deno.makeTempDir();
  try {
    await test();
  } finally {
    await Deno.remove(tempDirPath, { recursive: true });
  }
}
```

Then doing:

```ts
Deno.test('.....', withTempDir(async (tempDirPath) => {
  // ...test goes here...
});
```